core/src/main/java/feast/core/model/DataSource.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import static feast.proto.core.DataSourceProto.DataSource.SourceType.*;
+
+import feast.core.util.TypeConversion;
+import feast.proto.core.DataSourceProto;
+import feast.proto.core.DataSourceProto.DataSource.BigQueryOptions;
+import feast.proto.core.DataSourceProto.DataSource.FileOptions;
+import feast.proto.core.DataSourceProto.DataSource.KafkaOptions;
+import feast.proto.core.DataSourceProto.DataSource.KinesisOptions;
+import feast.proto.core.DataSourceProto.DataSource.SourceType;
+import java.util.HashMap;
+import java.util.Map;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter(AccessLevel.PRIVATE)
+@Table(name = "data_sources")
+public class DataSource {
+  @Column(name = "id")
+  @Id
+  @GeneratedValue
+  private long id;
+
+  // Type of this Data Source
+  @Enumerated(EnumType.STRING)
+  @Column(name = "type", nullable = false)
+  private SourceType type;
+
+  // DataSource Options
+  @Column(name = "config")
+  private String configJSON;
+
+  // Field mapping between sourced fields (key) and feature fields (value).
+  // Stored as serialized JSON string.
+  @Column(name = "field_mapping", columnDefinition = "text")
+  private String fieldMapJSON;
+
+  @Column(name = "ts_column")
+  private String tsColumn;
```

Comment:
should we make this `timestamp_column` maybe?

Reply:
Replace all ts_column/tsColumn with timestamp_column/timestampColumn respectively.